dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.h

```diff
@@ -64,6 +65,47 @@ typedef RcHandle<RtpsUdpInst> RtpsUdpInst_rch;
 typedef RcHandle<RtpsUdpTransport> RtpsUdpTransport_rch;
 typedef RcHandle<TransportClient> TransportClient_rch;
 
+struct LocatorCacheKey {
+  LocatorCacheKey(const RepoId& remote, const RepoId& local, bool prefer_unicast) : remote_(remote), local_(local), prefer_unicast_(prefer_unicast) {}
+  bool operator<(const LocatorCacheKey& rhs) const {
+    return std::memcmp(this, &rhs, sizeof (LocatorCacheKey)) < 0;
```

Comment:
This assumes that the struct is the same as if it were packed. Is there a cleaner way you could do this?

Reply:
Hmm, I suppose I am assuming it's packed here, so I ought to make that explicit.

Reply:
Can it delegate to `GUID_t`'s `operator<`?  The compiler should inline.

Reply:
I'm inclined to expand it to a single memcmp to avoid multiple function calls...